compiler/src/dotty/tools/dotc/reporting/WConf.scala

```diff
@@ -21,11 +20,15 @@ enum MessageFilter:
       val noHighlight = message.msg.message.replaceAll("\\e\\[[\\d;]*[^\\d;]","")
       pattern.findFirstIn(noHighlight).nonEmpty
     case MessageID(errorId) => message.msg.errorId == errorId
+    case SourcePattern(pattern) =>
+      val path = message.position.orElse(NoSourcePosition).source().path()
```

Comment:
```suggestion
      val path = message.position.orElse(NoSourcePosition).source.file.jpath.toAbsolutePath.toUri.normalize.getRawPath
```

It's a mouthful.. I guess build tools pass an absolute path already when invoking the compiler, but it's probably still good to use `toAbsolutePath`. Going through `toUri.getRawPath` avoids backslashes on Windows, `normalize` removes `/..` and alike.